clients/java/signalr/src/main/java/com/microsoft/aspnet/signalr/HubConnection.java

```diff
@@ -199,12 +203,12 @@ public CompletableFuture start() throws Exception {
     /**
      * Stops a connection to the server.
      */
-    private void stop(String errorMessage) {
+    private CompletableFuture<Void> stop(String errorMessage) {
```

Comment:
Should this happen in a `runAsync` call so that users could fire and forget the completion of what ever onClosed callbacks they have?

Reply:
Mostly ignore this part as I changed it again in the HttpClient change

Reply:
Ah, okay. Yeah, I'm going over that now